tasks/species_typing/task_srst2_vibrio.wdl

```diff
@@ -0,0 +1,159 @@
+          for row in reader:
+              result_list.append(dict(row))
+      # only one sample is run, so there's only one row, flattening list
+      return result_list[0]
+
+    # Converting None to empty string
+    conv = lambda i : i or '-'
+
+    # Make characters human-readable 
+    def translate_chars(string):
+      translation = []
+      if '?' in string:
+        translation.append("low depth/uncertain")
+      if '-' in string:
+        translation.append("not detected")
+      
+      # in case we want to retrieve the allele information
+      string = re.sub("\*|\?|-", "", string)
+
+      if len(translation) > 0:
+        return '(' + ';'.join(translation) + ')'
+      return ""
+
+    # load output TSV as dict 
+    row = tsv_to_dict('~{samplename}.tsv')
+  
+    # presence or absence genes - ctxA, ompW and toxR
+    with open("ctxA", "wb") as ctxA_fh:
+      value = row.get("ctxA")
+      presence = translate_chars(conv(value))
+      if presence == "(not detected)":
+        ctxA_fh.write(presence)
+      else:
+        result = "present" + ' ' + presence
+        ctxA_fh.write(result.strip())
+    
+    with open("ompW", "wb") as ompW_fh:
+      value = row.get("ompW")
+      presence = translate_chars(conv(value))
+      if presence == "(not detected)":
+        ompW_fh.write(presence)
+      else:
+        result = "present" + ' ' + presence
+        ompW_fh.write(result.strip())
+    
+    with open("toxR", "wb") as toxR_fh:
+      value = row.get("toxR")
+      presence = translate_chars(conv(value))
+      if presence == "(not detected)":
+        toxR_fh.write(presence)
+      else:
+        result = "present" + ' ' + presence
+        toxR_fh.write(result.strip())
+    
+    # biotype - tcpA classical or tcpA ElTor
+    with open("BIOTYPE", "wb") as biotype_fh:
+      value_ElTor = translate_chars(conv(row.get("tcpA_ElTor")))
+      value_classical = translate_chars(conv(row.get("tcpA_classical")))
+
+      if value_ElTor == "(not detected)" and value_classical == "(not detected)":
+        biotype_fh.write("(not detected)")
+      else:
+        if value_ElTor == "(not detected)":
+          result = "tcpA_Classical" + ' ' + value_classical
+          biotype_fh.write(result.strip())
+        else:
+          result = "tcpA_ElTor" + ' ' + value_ElTor
+          biotype_fh.write(result.strip())
+        
+    # serogroup - O1 or O139
+    with open("SEROGROUP", "wb") as serotype_fh:
+      value_O1 = translate_chars(conv(row.get("wbeN_O1")))
+      value_O139 = translate_chars(conv(row.get("wbfR_O139")))
+
+      if value_O1 == "(not detected)" and value_O139 == "(not detected)":
+        serotype_fh.write("(not detected)")
+      else:
+        if value_O1 == "(not detected)":
+          result = "O139" + ' ' + value_O139
+          serotype_fh.write(result.strip())
+        else:
+          result = "O1" + ' ' + value_O1
+          serotype_fh.write(result.strip())
+    CODE
+  >>>
```

Comment:
So good! For future iterations, I'm thinking that it may be helpful to package all of this gene-type parsing into a single executable that can be made available in the docker image itself to promote interoperability of this specific functionality.